### PR TITLE
build image from source using multi-stage

### PR DESCRIPTION
**Pull request by xinity**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,19 @@
...
+
+# install dependencies
+RUN apk add --no-cache glide make git
+
+# clone noderig repository and build binary
+
+RUN git clone https://github.com/ovh/noderig.git $GOPATH/src/github.com/ovh/noderig \
+    && cd $GOPATH/src/github.com/ovh/noderig \
+    && glide install \
+    && go build noderig.go
```
Comment (FlorentinDUBOIS) [inline Dockerfile]: You should use `make release` in order to compile noderig

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+
+# final stage
+FROM alpine:3.7
+COPY --from=build-env /go/src/github.com/ovh/noderig/noderig /
```
Comment (FlorentinDUBOIS) [inline Dockerfile]: With `make release` the binary is located under `build` directory. So, this command comes:

```
COPY --from=build-env /go/src/github.com/ovh/noderig/build/noderig /
```

Furthermore, I think it is better to store the binary under `/usr/local/bin`. The command will be:

```
COPY --from=build-env /go/src/github.com/ovh/noderig/build/noderig /usr/local/bin/
```

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,19 @@
...
 EXPOSE 9100
-ENTRYPOINT ["/usr/bin/noderig"]
-COPY noderig /usr/bin/noderig
+ENTRYPOINT ["/fossil"]
```
Comment (FlorentinDUBOIS) [inline Dockerfile]: It is not the good binary. You should use `/usr/local/bin/noderig`.

**Comment by xinity** [comment]
Will fix thanks for the review

**Comment by d33d33** [comment]
@xinity any news on this one?

**Comment by xinity** [comment]
@d33d33 better late than never :(
is this still relevant to fix ?

if so let me know and i'll make sure to fix this ASAP

**Pull request closed without merging**